book/garbage-collector/README.md

```diff
@@ -442,17 +459,65 @@ Let's see this for ourselves with a simple test program. You'll need to
 install the Core benchmarking suite via `opam install core_bench` before you
 compile this code:
 
-<link rel="import" href="code/gc/barrier_bench/barrier_bench.ml" />
+```ocaml file=../../examples/code/gc/barrier_bench/barrier_bench.ml
+open Core
+open Core_bench
+
+type t1 = { mutable iters1: int; mutable count1: float }
+type t2 = { iters2: int; count2: float }
+
+let rec test_mutable t1 =
+  match t1.iters1 with
+  |0 -> ()
+  |_ ->
+    t1.iters1 <- t1.iters1 - 1;
+    t1.count1 <- t1.count1 +. 1.0;
+    test_mutable t1
+
+let rec test_immutable t2 =
+  match t2.iters2 with
+  |0 -> ()
+  |n ->
+    let iters2 = n - 1 in
+    let count2 = t2.count2 +. 1.0 in
+    test_immutable { iters2; count2 }
+
+let () =
+  let iters = 1000000 in
+  let tests = [
+    Bench.Test.create ~name:"mutable"
+      (fun () -> test_mutable { iters1=iters; count1=0.0 });
+    Bench.Test.create ~name:"immutable"
+      (fun () -> test_immutable { iters2=iters; count2=0.0 })
+  ] in
+  Bench.make_command tests |> Command.run
+```
 
 This program defines a type `t1` that is mutable and `t2` that is immutable.
 The benchmark loop iterates over both fields and increments a counter.
 Compile and execute this with some extra options to show the amount of
 garbage collection occurring:
 
-<link rel="import" href="code/gc/barrier_bench/dune" />
+```scheme
```

Comment:
Not a blocker for this PR, but given that this is an OCaml tool, it might be nice to explicitly call this `dune` rather than `scheme`.  That would let us do dune-specific parsing in the future such as link to the online docs for particular stanzas.